xdist adds current directory to sys.path

With 1.26.1.
```python
import sys

def test_path():
    assert "" not in sys.path
```
```
$ pytest test_path.py
<OK>
$ pytest test_path.py -n 1
E     AssertionError....
```
Obviously this can cause problems/change behaviour depending on where you are running the tests from.

Possibly related to #376?

Yes, Initially, but the error (path turning into `''` therefore current directory import) happens in 3 also, either a virtualenv or a simple docker:
```dockerfile
FROM python:3
RUN pip3 install -U pytest-xdist==1.26.1
RUN printf "import sys\ndef test_path():\n  assert not '' in sys.path\n" >> test_path.py
RUN pytest test_path.py
RUN pytest test_path.py -n 1
```
So, no `conftest.py` or `site.py` or anything interfering. Results in:
```
Step 4/5 : RUN pytest test_path.py
============================= test session starts ==============================
platform linux -- Python 3.7.2, pytest-4.3.1, py-1.8.0, pluggy-0.9.0
rootdir: /, inifile:
plugins: xdist-1.26.1, forked-1.0.2
collected 1 item

test_path.py .                                                           [100%]

=========================== 1 passed in 0.01 seconds ===========================
```
```
Step 5/5 : RUN pytest test_path.py -n 1
============================= test session starts ==============================
platform linux -- Python 3.7.2, pytest-4.3.1, py-1.8.0, pluggy-0.9.0
rootdir: /, inifile:
plugins: xdist-1.26.1, forked-1.0.2
gw0 I
gw0 [1]

F                                                                        [100%]
=================================== FAILURES ===================================
__________________________________ test_path ___________________________________
[gw0] linux -- Python 3.7.2 /usr/local/bin/python

    def test_path():
>     assert not '' in sys.path
E     AssertionError: assert not '' in ['/', '', '/usr/local/lib/python37.zip', '/usr/local/lib/python3.7', '/usr/local/lib/python3.7/lib-dynload', '/usr/local/lib/python3.7/site-packages']
E      +  where ['/', '', '/usr/local/lib/python37.zip', '/usr/local/lib/python3.7', '/usr/local/lib/python3.7/lib-dynload', '/usr/local/lib/python3.7/site-packages'] = sys.path

test_path.py:3: AssertionError
=========================== 1 failed in 0.32 seconds ===========================
The command '/bin/sh -c pytest test_path.py -n 1' returned a non-zero code: 1
```
It looks to be the same root cause though.

the problem is the way execnet runs python, vs setuptools

when you run with `/home/graingert/projects/pytest-bdd/.tox/py39-pytestlatest-xdist/bin/pytest` the sys.path is set to:

```
E         +  '/home/graingert/projects/pytest-bdd/.tox/py39-pytestlatest-xdist/bin',
E         +  '/usr/lib/python39.zip',
E         +  '/usr/lib/python3.9',
E         +  '/usr/lib/python3.9/lib-dynload',
E         +  '/home/graingert/projects/pytest-bdd/.tox/py39-pytestlatest-xdist/lib/python3.9/site-packages',
```

however execnet runs with 

`/home/graingert/projects/pytest-bdd/.tox/py39-pytestlatest-xdist/bin/python`

and so the path is:

```
['', '/usr/lib/python39.zip', '/usr/lib/python3.9', '/usr/lib/python3.9/lib-dynload', '/home/graingert/projects/pytest-bdd/.tox/py39-pytestlatest-xdist/lib/python3.9/site-packages']
```

--- a/changelog/421.bugfix.rst
+++ b/changelog/421.bugfix.rst
@@ -1,0 +1,1 @@
+Copy the parent process sys.path into local workers, to work around execnet's python -c adding the current directory to sys.path.

--- a/src/xdist/plugin.py
+++ b/src/xdist/plugin.py
@@ -1,8 +1,11 @@
 import os
 import uuid
+import sys
 
 import py
 import pytest
+
+_sys_path = list(sys.path)  # freeze a copy of sys.path at interpreter startup
 
 
 def pytest_xdist_auto_num_workers():

--- a/src/xdist/remote.py
+++ b/src/xdist/remote.py
@@ -219,12 +219,14 @@ if __name__ == "__channelexec__":
     channel = channel  # noqa
     workerinput, args, option_dict, change_sys_path = channel.receive()
 
-    if change_sys_path:
+    if change_sys_path is None:
         importpath = os.getcwd()
         sys.path.insert(0, importpath)
         os.environ["PYTHONPATH"] = (
             importpath + os.pathsep + os.environ.get("PYTHONPATH", "")
         )
+    else:
+        sys.path = change_sys_path
 
     os.environ["PYTEST_XDIST_TESTRUNUID"] = workerinput["testrunuid"]
     os.environ["PYTEST_XDIST_WORKER"] = workerinput["workerid"]

--- a/src/xdist/workermanage.py
+++ b/src/xdist/workermanage.py
@@ -9,6 +9,7 @@ import pytest
 import execnet
 
 import xdist.remote
+from xdist.plugin import _sys_path
 
 
 def parse_spec_config(config):
@@ -261,7 +262,8 @@ class WorkerController:
         remote_module = self.config.hook.pytest_xdist_getremotemodule()
         self.channel = self.gateway.remote_exec(remote_module)
         # change sys.path only for remote workers
-        change_sys_path = not self.gateway.spec.popen
+        # restore sys.path from a frozen copy for local workers
+        change_sys_path = _sys_path if self.gateway.spec.popen else None
         self.channel.send((self.workerinput, args, option_dict, change_sys_path))
 
         if self.putevent:

--- a/testing/test_remote.py
+++ b/testing/test_remote.py
@@ -234,7 +234,7 @@ def test_remote_env_vars(testdir):
 
 
 def test_remote_inner_argv(testdir):
-    """Work around sys.path differences due to execnet using `python -c`."""
+    """Test/document the behavior due to execnet using `python -c`."""
     testdir.makepyfile(
         """
         import sys
@@ -295,7 +295,7 @@ def test_remote_usage_prog(testdir, request):
 
 
 def test_remote_sys_path(testdir):
-    """Test/document the behavior due to execnet using `python -c`."""
+    """Work around sys.path differences due to execnet using `python -c`."""
     testdir.makepyfile(
         """
         import sys

--- a/testing/test_remote.py
+++ b/testing/test_remote.py
@@ -234,7 +234,7 @@ def test_remote_env_vars(testdir):
 
 
 def test_remote_inner_argv(testdir):
-    """Test/document the behavior due to execnet using `python -c`."""
+    """Work around sys.path differences due to execnet using `python -c`."""
     testdir.makepyfile(
         """
         import sys
@@ -292,3 +292,17 @@ def test_remote_usage_prog(testdir, request):
     result = testdir.runpytest_subprocess("-n1")
     assert result.ret == 1
     result.stdout.fnmatch_lines(["*usage: *", "*error: my_usage_error"])
+
+
+def test_remote_sys_path(testdir):
+    """Test/document the behavior due to execnet using `python -c`."""
+    testdir.makepyfile(
+        """
+        import sys
+
+        def test_sys_path():
+            assert "" not in sys.path
+        """
+    )
+    result = testdir.runpytest("-n1")
+    assert result.ret == 0

